kernel/include/kernel_internal.h

```diff
@@ -45,6 +46,51 @@ extern void z_setup_new_thread(struct k_thread *new_thread,
 			      void *p1, void *p2, void *p3,
 			      int prio, u32_t options, const char *name);
 
+static ALWAYS_INLINE void z_new_thread_init(struct k_thread *thread,
```

Comment:
could just put this in thread.c, doesn't need to be an inline

Reply:
will add a new commit addressing that.